tasks/sep/ink-004-enable-permissionless-game/SignFromJson.s.sol

```diff
@@ -0,0 +1,73 @@
+
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {SignFromJson as OriginalSignFromJson} from "script/SignFromJson.s.sol";
+import {Simulation} from "@base-contracts/script/universal/Simulation.sol";
+import {OptimismPortal2, IDisputeGame} from "@eth-optimism-bedrock/src/L1/OptimismPortal2.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/libraries/Types.sol";
+import {Vm, VmSafe} from "forge-std/Vm.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {stdToml} from "forge-std/StdToml.sol";
+import {LibString} from "solady/utils/LibString.sol";
+import {GnosisSafe} from "safe-contracts/GnosisSafe.sol";
+import "@eth-optimism-bedrock/src/dispute/lib/Types.sol";
+
+contract SignFromJson is OriginalSignFromJson {
+    using LibString for string;
+
+    // Chains for this task.
+    string constant l1ChainName = "sepolia";
+    string l2ChainName = vm.envString("L2_CHAIN_NAME");
+
+    Types.ContractSet proxies;
+
+    /// @notice Sets up the contract
+    function setUp() public {
+        proxies = _getContractSet();
+    }
+
+    function checkRespectedGameType() internal view {
+        OptimismPortal2 portal = OptimismPortal2(payable(proxies.OptimismPortal));
+        require(portal.respectedGameType().raw() == GameTypes.CANNON.raw());
+    }
+
+    function getAllowedStorageAccess() internal view override returns (address[] memory allowed) {
+        allowed = new address[](2);
+        allowed[0] = proxies.OptimismPortal;
+        allowed[1] = vm.envAddress("OWNER_SAFE");
+    }
+
+    /// @notice Checks the correctness of the deployment
+    function _postCheck(Vm.AccountAccess[] memory accesses, Simulation.Payload memory /* simPayload */ )
+        internal
+        view
+        override
+    {
+        console.log("Running post-deploy assertions");
+
+        checkStateDiff(accesses);
+        checkRespectedGameType();
+
+        console.log("All assertions passed!");
+    }
+
+    /// @notice Reads the contract addresses from lib/superchain-registry/superchain/configs/${l1ChainName}/${l2ChainName}.toml
+    function _getContractSet() internal view returns (Types.ContractSet memory _proxies) {
+        string memory chainConfig;
+
+        // Read chain-specific config toml file
+        // Unichain Mainnet is not yet public, so there is a local toml file included in this task
```

Comment:
Looks like this was taken from unichain sepolia, we should update this to use the SuperchainRegistry base contract and load this stuff from the SuperchainRegistry